net/net.go

```diff
@@ -16,7 +16,71 @@ limitations under the License.
 
 package net
 
-import "net"
+import (
+	"fmt"
+	"net"
+)
+
+// ParseCIDRsStrings parses a list of cidrs and return error if any is in valid.
+// order is maintained
+func ParseCIDRsStrings(cidrsString []string) ([]*net.IPNet, error) {
+	cidrs := make([]*net.IPNet, len(cidrsString))
```

Comment:
It is preferred to not do this unless it's really a perf issue, so we don't have to deal with indexing:
```
var cidrs []*net.IPNet
for _, s := range cidrsString {
  ...
  cidrs = append(cidrs, cidr)
}
```